docker-k8s.yml

```diff
@@ -0,0 +1,275 @@
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: init-db
+spec:
+  selector:
+    matchLabels:
+      name: init-db
+  template:
+    metadata:
+      labels:
+        name: init-db
+    spec:
+      containers:
+      - image: dashersw/cote-workshop
+        name: init-db
+        command: ["node", "init-db.js"]
+        env:
+          - name: PG
+            value: "true"
+          - name: COTE_MULTICAST_ADDRESS
+            value: '239.1.11.131'
+---
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: admin
+spec:
+  selector:
+    matchLabels:
+      name: admin
+  template:
+    metadata:
+      labels:
+        name: admin
+    spec:
+      containers:
+      - image: dashersw/cote-workshop
+        name: admin
+        command: ["node", "admin/server.js"]
+        ports:
+        - containerPort: 5000
+        env:
+          - name: PG
+            value: "true"
+          - name: COTE_MULTICAST_ADDRESS
+            value: '239.1.11.131'
+---
+
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: end-user
+spec:
+  selector:
+    matchLabels:
+      name: end-user
+  template:
+    metadata:
+      labels:
+        name: end-user
+    spec:
+      containers:
+      - image: dashersw/cote-workshop
+        name: admin
+        command: ["node", "end-user/server.js"]
+        ports:
+        - containerPort: 5001
+        env:
+          - name: PG
+            value: "true"
+          - name: COTE_MULTICAST_ADDRESS
+            value: '239.1.11.131'
```

Comment:
is there something like a variable in k8s templates? there's a lot of repetition in the file, including but not limited to the multicast address.

Reply:
Yes k8s support configmaps which can share configs such as these between deployments.  I didn't include them as I tried to limit things we were using. 

As for the duplication of these services we could use yaml achnors or standard templates which most people do anyway.